src/Castle.Windsor/MicroKernel/Context/CreationContext.cs

```diff
@@ -321,7 +321,11 @@ private bool CanConvertParameter(Type type)
 		private bool CanResolve(DependencyModel dependency, object inlineArgument)
 		{
 			var type = dependency.TargetItemType;
-			if (inlineArgument == null || type == null)
+			if (inlineArgument == null)
+			{
+				return type.IsNullable();
```

Comment:
Could `type` be null here?

Reply:
Good find, thanks!

Reply:
I haven't been able to figure out how to write a test that created a dependency with a null type, which is what would be required. This code creates a DependencyModel with a null type:

`Component.For(typeof(ClassWithListConstructor)).Named("key")` 

But I'm not sure how to go from there to making the DependencyModel be a dependency of another component.

Reply:
I'll just fix without a test unless you know where to point me.

Reply:
Sounds good, you've covered a good number of scenarios, thanks!